Configure.cmake

```diff
@@ -42,6 +45,28 @@ CHECK_C_SOURCE_COMPILES("
     __m128d r = _mm_mul_pd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
   COMPILER_SUPPORTS_SSE2)
 
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_SSE4})
+CHECK_C_SOURCE_COMPILES("
+  #if defined(_MSC_VER)
+  #include <intrin.h>
+  #else
+  #include <x86intrin.h>
+  #endif
+  int main() {
+    __m128d r = _mm_floor_sd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
+  COMPILER_SUPPORTS_SSE4)
+
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_AVX})
```

Comment:
I think there is no right or wrong, but it's a matter of what we decide. From what I understand, the question is: 
_Should the library build symbols that the native host does not support?_

If we currently build AVX512 on systems that can't run it, then I see no problem in doing the same for AVX on Cygwin. 
There is the other take that maybe it's not a good idea to provide vector symbols for architectures that the _native_ host system does not support, but only that for cross-compiled builds. 

I think we need to decide and be consistent, but I might have misunderstood and need some clarification? @shibatch